lib/plugins/bodyReader.js

```diff
@@ -172,6 +172,7 @@ function bodyReader(options) {
         });
 
         req.once('error', next);
+        req.once('close', next);
```

Comment:
Could you please wrap the next function here in a `once`? We're about to tidy up the router and server such that it's going to enforce strict next, and so we would want to make sure we're not potentially calling next here.

If you can also add some comments about why we're adding the extra listener here, given the context you have about the memory leak, that would be even more awesome.

Reply:
We may have to handle the `aborted` event as well.

Reply:
Yeah, I think handling aborted is a good idea.  I was looking at the ReadableStream docs, but it appears ClientRequest (the req object) adds a couple of additional events.

Reply:
Should I wrap all the next calls in bodyReader w/ once?

Reply:
You can either do that, or when next fires deregister all the other listeners on the request that you've previously attached.